File: Leet_1828/solution.py
class Solution:
    def countPoints(self, points: list[list[int]], queries: list[list[int]]) -> list[int]:
        result: list[int] = []

        for query in queries:
            count: int = 0

            for point in points:
                # check point is inside circle
                if pow(point[0] - query[0], 2) + pow(point[1] - query[1], 2) <= pow(query[2], 2):
                    count += 1

            result.append(count)

        return result
    

def main() -> None:
    print("1828. Queries on Number of Points Inside a Circle")
    sol = Solution()
    print(sol.countPoints([[1,3],[3,3],[5,3],[2,2]], [[2,3,1],[4,3,1],[1,1,2]]))
    print(sol.countPoints([[1,1],[2,2],[3,3],[4,4],[5,5]], [[1,2,2],[2,2,2],[4,3,2],[4,3,3]]))


if __name__ == "__main__":
    main()

File: Leet_1828/solution.c
int* countPoints(int** points, int pointsSize, int* pointsColSize,
                 int** queries, int queriesSize, int* queriesColSize,
                 int* returnSize) {
    int size = 0;
    int rSize = pointsSize;
    int* result = calloc(queriesSize, sizeof(int));

    for (int i = 0; i < queriesSize; i++) {
        int count = 0;
        
        /* NOTE check pointsColSize */
        int originX = queries[i][0];
        int originY = queries[i][1];
        int radius = queries[i][2];

        for (int j = 0; j < pointsSize; j++) {

            /* NOTE check queriesColSize */
            int pointX = points[j][0];
            int pointY = points[j][1];

            int x2 = (pointX - originX) * (pointX - originX);
            int y2 = (pointY - originY) * (pointY - originY);

            /* check point is inside circle */
            if (x2 + y2 <= radius * radius) {
                count++;
            }
        }

        result[size++] = count;
        *returnSize = size;
    }

    return result;
}
